testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jsonb/basic/JsonBindingDebugLoggingTest.java

```diff
@@ -0,0 +1,325 @@
+   }
+
+   /**
+    * @tpTestDetails Check exception during server sending
+    * @tpSince RESTEasy 4.0.0.Beta7
+    *
+    * In this scenario the exception originates in JsonBindingProvider.writeTo.
+    * The exception is passed up the call stack to SynchronousDispatcher.writeException.
+    * The ExceptionHandler processes the original exception and identifies it as an
+    * UnhandledException.  The exception is written to the server.log by this class
+    * and the original exception is wrapped in an UnhandledException class and returned
+    * up the call stack.  This exception is passed through undertow code which adds
+    * HTML formatting to the message that is returned to the caller.
+    */
+   @Test
+   public void exceptionDuringServerSend() throws Exception {
+      // count log messages before request
+      LogCounter errorStringLog = new LogCounter("ERROR",
+              false, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      LogCounter resteasyExceptionLog = new LogCounter(
+              ".*ERROR .* RESTEASY002025.*", true,
+              ContainerConstants.DEFAULT_CONTAINER_QUALIFIER, true);
+
+      LogCounter yassonExceptionLog = new LogCounter(
+              "Caused by: javax.json.bind.JsonbException",
+              true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter yassonStacktraceLog = new LogCounter(
+              "at org.eclipse.yasson", true,
+              ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      LogCounter applicationExcpetionLog = new LogCounter(
+              "Caused by: java.lang.RuntimeException: "
+            + JsonBindingDebugLoggingItemCorruptedGet.class.getSimpleName(),
+              true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      // perform request
+      WebTarget base = client.target(generateURL("/get/nok"));
+      Response response = base.request().get();
+
+      // check response
+      Assert.assertThat("Wrong response code", response.getStatus(), is(500));
+      Assert.assertThat("Response message doesn't contains full stacktrace",
+              response.readEntity(String.class), allOf(
+            containsString("org.eclipse.yasson.internal"),
+            containsString("java.lang.RuntimeException: "
+                              + JsonBindingDebugLoggingItemCorruptedGet.class.getSimpleName()),
+            containsString("javax.json.bind.JsonbException: Error getting value on"),
+            containsString("RESTEASY008205")
+      ));
+
+      // checking server.log contents
+      // For applicationExcpetionLog and yassonExceptionLog there are
+      // always 2 log records for this text.  One is generated from
+      // resteasy and one is generated from undertow.
+      // assert log messages after request
+      Assert.assertThat("Application Exception should be logged",
+              applicationExcpetionLog.count(), is(2));

```

Comment:
Why twice? My original commit contains this:
```java
Assert.assertThat("Application Exception should be logged", applicationExcpetionLog.count(), is(2)); // TBD replace to 1
```

Reply:
Because it show up twice. Once from resteasy and once from undertow.